src/cli/test/task-generate.spec.ts

```diff
@@ -0,0 +1,137 @@
+  const tmp = console.log;
+  const log = jest.fn();
+  console.log = log;
+  await taskGenerate(coreCompiler, config);
+  console.log = tmp;
+  return log;
+}
+
+describe('generate task', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+    jest.clearAllMocks();
+    jest.resetModules();
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should exit with an error if no `configPath` is supplied', async () => {
+    const { config, errorSpy } = await setup();
+    config.configPath = undefined;
+    await taskGenerate(coreCompiler, config);
+    expect(config.sys.exit).toBeCalledWith(1);
+    expect(errorSpy).toBeCalledWith(
+      'Please run this command in your root directory (i. e. the one containing stencil.config.ts).'
+    );
+  });
+
+  it('should exit with an error if no `srcDir` is supplied', async () => {
+    const { config, errorSpy } = await setup();
+    config.srcDir = undefined;
+    await taskGenerate(coreCompiler, config);
+    expect(config.sys.exit).toBeCalledWith(1);
+    expect(errorSpy).toBeCalledWith("Stencil's srcDir was not specified.");
+  });
+
+  it('should exit with an error if the component name does not validate', async () => {
+    const { config, errorSpy, validateTagSpy } = await setup();
+    validateTagSpy.mockReturnValue('error error error');
+    await taskGenerate(coreCompiler, config);
+    expect(config.sys.exit).toBeCalledWith(1);
+    expect(errorSpy).toBeCalledWith('error error error');
+  });
+
+  it.each([true, false])('should create a directory for the generated components', async (includeTests) => {
+    const { config } = await setup();
+    if (!includeTests) {
+      promptMock.mockResolvedValue({
+        tagName: 'my-component',
+        // simulate the user picking only the css option
+        filesToGenerate: ['css'],
+      });
+    }
+
+    const createDirSpy = jest.spyOn(config.sys, 'createDir');
+    await silentGenerate(coreCompiler, config);
+    expect(createDirSpy).toBeCalledWith(
+      includeTests ? `${config.srcDir}/components/my-component/test` : `${config.srcDir}/components/my-component`,
+      { recursive: true }
+    );
+  });
+
+  it('should generate the files the user picked', async () => {
+    const { config } = await setup();
+    const writeFileSpy = jest.spyOn(config.sys, 'writeFile');
+    await silentGenerate(coreCompiler, config);
+    [
+      { extension: 'tsx', path: '/src/components/my-component/my-component.tsx' },
+      { extension: 'css', path: '/src/components/my-component/my-component.css' },
+      { extension: 'spec.tsx', path: '/src/components/my-component/test/my-component.spec.tsx' },
+      { extension: 'e2e.ts', path: '/src/components/my-component/test/my-component.e2e.ts' },
+    ].forEach((file) => {
+      expect(writeFileSpy).toBeCalledWith(
+        file.path,
+        getBoilerplateByExtension('my-component', file.extension as GenerableExtension, true)
```

Comment:
I think I had something similar to that in a version of this test that I changed before pushing, actually because I didn't want to introduce too many new exports. If that's alright though I'm down with that change, I like avoiding `as` whenever I can